### DISCUSSION TITLE
How to combine web history and memory history?

### DISCUSSION THREAD

**Discussion by bolau** [nuxt/nuxt#21029] [Questions]
Hi, 
I'm working on a game in the form of a single page app based on Nuxt 3. 

For showing different screens in the actual game, I'd like to avoid exposing the history and a back function to the user. I understand, that using pages with a memory-based history might be the way to go here.
For additional pages like imprint, data protection, help, faq and so on, however, I'd like to use pages with a regular web-based history.

What can I do?
- Is there a way to use a memory history for a subfolder, while using a web history for the rest?
- Is using pages with a nested `NuxtPage` the right thing to do to switch between the different screens of the game, or is there another concept?

Thanks for any help
Boris

**Comment by maou-shonen** [comment]
You can't simply combine two Vue routers.
If you don't have many game pages, I think using router.replace for navigation within those pages would meet your needs.
If you are certain about using a particular strategy, you can do something in onBeforeRouteUpdate in Vue Router.

**Reply by bolau** [reply]
Thanks for the reply. Replacing the current page in the history prevents going back, but it does make the URL change. I'll look into onBeforeRouteUpdate and see what it can do.

**Reply by maou-shonen** [reply]
Then you can also consider using dynamic components.

**Comment by bolau** [comment]
I found a hacky way to get what I'm looking for:
- use a memory history
- use `<a href=...` to link to the meta pages, and `<NuxtLink to=...` to link between the game screens

This is probably considered "off-label-use", and I wonder how future-proof it is. But it works for now.
Thanks for the suggestions!

**Reply by maou-shonen** [reply]
Using <a> tags for navigation will cause Vue to be reinitialized every time.
if the page transitions are not frequent...it can be a workaround.
only you know if this approach suits your needs 😃
